src/app/main.rs

```diff
@@ -337,6 +337,7 @@ where
                 let endpoint_stack = client_stack
                     .push(buffer::layer(MAX_IN_FLIGHT))
                     .push(settings::router::layer::<Endpoint, _>())
+                    .push(server_id::layer())
```

Comment:
I'd suggest that we explicitly clear the header "before" setting it in this layer, so there's no potential that the remote could have sent it to forge the proxy's value. Unless there's something else preventing this?

Reply:
Done.